client/client.cpp

```diff
@@ -63,21 +63,27 @@ void init() {
   client::BW::data::init();
 }
 
+//============================= LIFECYCLE ====================================
+
 Client::Client() : state_(new State()) {}
 
 Client::~Client() {
   state_->decref();
 }
 
-bool Client::connect(const std::string& hostname, int port) {
+//============================= OPERATIONS ===================================
+
+bool Client::connect(const std::string& hostname, int port,
+    int send_timeout_ms /* = -1 */, int receive_timeout_ms /* = -1 */) {
```

Comment:
Since these are 2 separate settings in ZMQ, I wanted to keep them separate in TorchCraft as well. Finally, it's just on the connection setup that those are specified, so duplication does not take much effort. At the same time I cannot come up with a use case where they should be different.

As for the style, the field names in client.h use underscores (e.g. Client::Options::initial_map, window_pos, window_size etc); I'm more used to underscore convention in C++ (boost, STL) and camel case in Java. But it's just a matter of convention.